src/app/components/algorithms/implementation-view/dialogs/implementation-nisq-analyzer-qpu-selection-dialog/implementation-nisq-analyzer-qpu-selection-dialog.component.ts

```diff
@@ -0,0 +1,130 @@
+  ],
+})
+export class ImplementationNisqAnalyzerQpuSelectionDialogComponent
+  implements OnInit {
+  qpuSelectionFrom: FormGroup;
+  provider?: EntityModelProviderDto;
+  ready?: boolean;
+  isIbmqSelected = true;
+  isSimulatorAllowed = false;
+
+  constructor(
+    public dialogRef: MatDialogRef<
+      ImplementationNisqAnalyzerQpuSelectionDialogComponent
+    >,
+    @Inject(MAT_DIALOG_DATA) public data: DialogData,
+    public dialog: MatDialog,
+    private providerService: ProviderService
+  ) {}
+
+  get vendor(): AbstractControl | null {
+    return this.qpuSelectionFrom.get('vendor');
+  }
+
+  get simulatorAllowed(): AbstractControl | null {
+    return this.qpuSelectionFrom.get('simulatorAllowed');
+  }
+
+  get token(): AbstractControl | null {
+    return this.qpuSelectionFrom.get('token');
+  }
+
+  ngOnInit(): void {
+    this.qpuSelectionFrom = new FormGroup({
+      vendor: new FormControl(this.data.vendor, [
+        // eslint-disable-next-line @typescript-eslint/unbound-method
+        Validators.required,
+      ]),
+      simulatorAllowed: new FormControl(this.data.simulatorAllowed, [
+        // eslint-disable-next-line @typescript-eslint/unbound-method
+        Validators.required,
+      ]),
+      token: new FormControl(this.data.token, [
+        // eslint-disable-next-line @typescript-eslint/unbound-method
+        Validators.required,
+        Validators.maxLength(255),
+      ]),
+    });
+
+    this.vendor.setValue('IBMQ');
+    this.onVendorChanged(this.vendor.value);
+    this.simulatorAllowed.setValue(false);
+
+    this.dialogRef.beforeClosed().subscribe(() => {
+      this.data.vendor = this.vendor.value;
+      this.data.simulatorAllowed = this.simulatorAllowed.value;
+      this.data.token = this.token.value;
+    });
+  }
+
+  onNoClick(): void {
+    this.dialogRef.close();
+  }
+
+  isRequiredDataMissing(): boolean {
+    return this.vendor.errors?.required;
+    return this.token.errors?.required;
+  }
+
+  onVendorChanged(value: string): void {
+    this.isIbmqSelected = true;
+    if (value === 'IBMQ') {
+      this.providerService.getProviders().subscribe((result) => {
+        this.getProviderDtoByName(result);
+        if (!this.provider) {
+          console.error('Provider with given name not found!');
+          this.ready = true;
+          return;
+        }
+      });
+    } else {
+      this.isIbmqSelected = false;
+    }
+  }
+
+  getProviderDtoByName(result): void {
+    if (result === null) {
+      console.error('Error while loading provider!');
+      return;
+    }
+    for (const providerDto of result._embedded.providerDtoes) {
+      if (providerDto.name.toLowerCase() === 'ibmq') {
+        this.provider = providerDto;
+        return;
+      }
+    }
+  }
+
+  setSimulatorAllowed(allowed: boolean): void {
+    this.isSimulatorAllowed = allowed;
+    this.simulatorAllowed.setValue(this.isSimulatorAllowed);
+  }
+}
+
+export interface DialogData {
```

Comment:
If you need this interface somewhere else, put it into a separate file. Otherwise you don't need the export statement.

Reply:
This specific DialogData is only used in the context of this dialog.